signer/tests/devenv/devenv.rs

```diff
@@ -0,0 +1,626 @@
+    // Need also to add some other tx to ensure we get a different block hash
+    faucet.send_to(1002, &faucet.address);
+    let block_1b = faucet.generate_block();
+    assert_ne!(block_1a, block_1b);
+
+    let block_2b = faucet.generate_block();
+
+    // Let's wait for the block observer signal
+    let signal = signal_rx.recv();
+    let Ok(SignerSignal::Event(SignerEvent::BitcoinBlockObserved)) = signal.await else {
+        panic!("Not the right signal")
+    };
+
+    let (bitcoin_tip_fork, _) = db.get_chain_tips().await;
+    assert_eq!(block_2b, bitcoin_tip_fork.block_hash.into());
+    assert_eq!(
+        bitcoin_tip_original.block_height + 1,
+        bitcoin_tip_fork.block_height
+    );
+
+    let original_in_fork = db
+        .in_canonical_bitcoin_blockchain(&bitcoin_tip_fork, &bitcoin_tip_original)
+        .await
+        .unwrap();
+    assert!(!original_in_fork);
+
+    block_observer_handle.abort();
+    testing::storage::drop_db(db).await;
+}
+
+/// Test checking `getrawtransaction` behaviour after a fork
+#[test_log::test(tokio::test)]
+async fn getrawtransaction_simple_fork() {
+    // We don't really need a context, we just need the bitcoin client
+    let ctx = TestContext::builder()
+        .with_in_memory_storage()
+        .with_first_bitcoin_core_client()
+        .with_mocked_stacks_client()
+        .with_mocked_emily_client()
+        .build();
+    let bitcoin = ctx.get_bitcoin_client();
+
+    let (rpc, faucet) = regtest::initialize_blockchain_devenv();
+
+    // Start from a clean mempool
+    faucet.generate_block();
+
+    // Prepare the UTXO used by the tx
+    let utxo_outpoint = &faucet.send_to(10_000, &faucet.address);
+    faucet.generate_block();
+
+    let utxos = faucet.get_utxos(None);
+    let utxo = utxos.iter().find(|u| u.txid == utxo_outpoint.txid).unwrap();
+
+    // Create a tx spending the UTXO
+    let mut tx = Transaction {
+        version: Version::ONE,
+        lock_time: LockTime::ZERO,
+        input: vec![TxIn {
+            previous_output: utxo.outpoint(),
+            sequence: Sequence::ZERO,
+            script_sig: ScriptBuf::new(),
+            witness: Witness::new(),
+        }],
+        output: vec![TxOut {
+            value: Amount::from_sat(9_000),
+            script_pubkey: faucet.address.script_pubkey(),
+        }],
+    };
+    let input_index = 0;
+    let keypair = &faucet.keypair;
+    match faucet.address.address_type().unwrap() {
+        AddressType::P2wpkh => {
+            regtest::p2wpkh_sign_transaction(&mut tx, input_index, utxo, keypair)
+        }
+        AddressType::P2tr => {
+            regtest::p2tr_sign_transaction(&mut tx, input_index, &[utxo.clone()], keypair)
+        }
+        _ => unimplemented!(),
+    };
+    rpc.send_raw_transaction(&tx).unwrap();
+    let txid = &tx.compute_txid();
+
+    // The tx should be in mempool
+    let tx = bitcoin.get_tx(txid).await.unwrap().unwrap();
+    assert!(tx.confirmations.is_none());
```

Comment:
I'm still not sure why here `confirmations` is `None`, while after confirming and forking it then we get `Some(0)`.

Reply:
Added more checks after offline chat, IIUC:
 - if the tx is in mempool (ie, is still a valid tx in the current mempool), regardless of its history, then we get null confirmations and no block hash
 - if the tx is not in the mempool (eg, invalidated by another one), then if it was included in some forked block we get that block and the number of confirmations (0, assuming nobody built on top of that).